stix_shifter_modules/elastic_ecs/stix_transmission/connector.py

```diff
@@ -44,12 +52,44 @@ def ping_connection(self):
             else:
                 raise e
 
-    def create_results_connection(self, query, offset, length):
+    def get_pagesize(self):
+        response_txt = None
+        return_obj = dict()
+        try:
+            response = self.api_client.get_max_result_window()
+            return_obj = self._handle_errors(response, return_obj)
+            if (return_obj['success']):
+                response_json = json.loads(return_obj["data"])
+                max_result_windows = list()
+                if not (response_json is None):
+                    for index, item_json in response_json.items():
+                        if 'max_result_window' in item_json['settings']['index']:
+                            max_res_win = item_json['settings']['index']['max_result_window']
+                        elif 'max_result_window' in item_json['defaults']['index']:
+                            max_res_win = item_json['defaults']['index']['max_result_window']
+                        else:
+                            ErrorResponder.fill_error(item_json,
+                                                      message='max_result_window is not set in index: ' + str(index),
+                                                      connector=self.connector)
+                            self.logger.error('max_result_window is not set in index: ' + str(index))
+                        max_result_windows.append(int(max_res_win))
+                max_result_window = sorted(max_result_windows)[0] #return the smallest max_return_window in indices
+                return max_result_window
+        except Exception as e:
+            if response_txt is not None:
+                ErrorResponder.fill_error(return_obj, message='unexpected exception', connector=self.connector)
+                self.logger.error('can not parse response: ' + str(response_txt))
+            else:
+                raise e
+
+    def create_results_connection(self, query, offset, length, metadata=None):
         response_txt = None
         return_obj = dict()
 
         try:
-            response = self.api_client.run_search(query, offset, length)
+            # using search after API in ElasticSearch
+            # pass the last searched value in metadata argument, ignore offset argument
+            response = self.api_client.search_pagination(query, metadata, min(length, self.max_result_window))
```

Comment:
i would cast length into int() to make sure it doesn't fail at this call:` int(length)`

Reply:
passing length in CLI arguments fails at this point because it reads it as a string.